dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -46,10 +50,17 @@ public static ToolMetadata createFrom(Tool annotation, Bean<?> bean, AnnotatedMe
         String name = annotation.name().equals(Tool.ELEMENT_NAME) ? method.getJavaMember().getName() : annotation.name();
         String title = annotation.title().isEmpty() ? null : annotation.title();
         String description = annotation.description().isEmpty() ? null : annotation.description();
+        Map<String, ArgumentMetadata> arguments = getArgumentMap(method);
+        List<SpecialArgumentMetadata> specialArguments = getSpecialArgumentList(method);
 
         WrapBusinessError wrapAnnotation = method.getAnnotation(WrapBusinessError.class);
         List<Class<? extends Throwable>> businessExceptions = (wrapAnnotation != null) ? List.of(wrapAnnotation.value()) : Collections.emptyList();
         boolean returnsCompletionStage = CompletionStage.class.isAssignableFrom(method.getJavaMember().getReturnType());
+        SchemaRegistry sr = SchemaRegistry.get();
+
+        JsonObject inputSchema = sr.getToolInputSchema(method);
+        JsonObject outputSchema = annotation.structuredContent() ? sr.getToolOutputSchema(method) : null;
```

Comment:
This structured content check needs to be a little more complex and match the logic in `McpServlet.evaluateToolResponse`